src/app/wallet/v2/views/buy_sell_crypto/service_item.nim

```diff
@@ -0,0 +1,65 @@
+import json, strformat, chronicles
+
+include ../../../../../status/utils/[json_utils]
+
+logScope:
+  topics = "crypto-service"
+  
+type CryptoServiceItem* = object
+  name: string
+  description: string
+  fees: string
+  logoUrl: string
+  siteUrl: string  
+  hostname: string
+
+proc initCryptoServiceItem*(name, description, fees, logoUrl, siteUrl, 
+  hostname: string): CryptoServiceItem =
+
+  result.name = name
+  result.description = description
+  result.fees = fees
+  result.logoUrl = logoUrl
+  result.siteUrl = siteUrl
+  result.hostname = hostname
+
+proc initCryptoServiceItem*(jsonObject: JsonNode): CryptoServiceItem =
+
+  if (jsonObject.kind != JObject):
+      info "CryptoServiceItem initialization failed: JsonNode is not JObject"
+      return
+
+  discard jsonObject.getProp("name", result.name)
+  discard jsonObject.getProp("description", result.description)
+  discard jsonObject.getProp("fees", result.fees)
+  discard jsonObject.getProp("logoUrl", result.logoUrl)
+  discard jsonObject.getProp("siteUrl", result.siteUrl)
+  discard jsonObject.getProp("hostname", result.hostname)
+
+proc `$`*(self: CryptoServiceItem): string =
+  result = "CryptoServiceItem("
+  result &= fmt"name:{self.name}, "
+  result &= fmt"description:{self.description}, "
+  result &= fmt"fees:{self.fees}, "
+  result &= fmt"logoUrl:{self.logoUrl}, "
+  result &= fmt"siteUrl:{self.siteUrl}"
+  result &= fmt"hostname:{self.hostname}"
+  result &= ")"
+
+method getName*(self: CryptoServiceItem): string {.base.} =
```

Comment:
Q: why doing that rather than making the attribute public? is it to protect from write?

Reply:
@alaibe exactly, we must not make everything public. Same states for methods/procedures. Making properties public, adding `*`, may corrupt our logic letting someone to change that property form the outside. In general we should expose only methods/procedures, not properties, unless we have some simple structure/type and that typing would save some time.